synapse/lib/stormlib/cortex.py

```diff
@@ -0,0 +1,656 @@
+    async def _gtorName(self):
+        return self.info.get('name')
+
+    async def _storDesc(self, desc):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        desc = await s_stormtypes.tostr(desc)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'desc', desc)
+        self.info['desc'] = desc
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorDesc(self):
+        return self.info.get('desc')
+
+    async def _storRunas(self, runas):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        runas = await s_stormtypes.tostr(runas)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'runas', runas)
+        self.info['runas'] = runas
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorRunas(self):
+        return self.info.get('runas')
+
+    async def _storReadonly(self, readonly):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        readonly = await s_stormtypes.tobool(readonly)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'readonly', readonly)
+        self.info['readonly'] = readonly
+
+    @s_stormtypes.stormfunc(readonly=True)
+    def _ctorVars(self, path=None):
+        return HttpApiVars(self, path=path)
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorReadonly(self):
+        return self.info.get('readonly')
+
+    async def _storOwner(self, owner):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        if isinstance(owner, s_stormtypes.User):
+            info = await owner.value()
+            owner = info.get('iden')
+        else:
+            owner = await s_stormtypes.tostr(owner)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'owner', owner)
+        self.info['owner'] = owner
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorOwner(self):
+        iden = self.info.get('owner')
+        udef = await self.runt.snap.core.getUserDef(iden)
+        if udef is None:
+            raise s_exc.NoSuchUser(mesg=f'HTTP API owner does not exist {iden}', user=iden)
+        return s_stormtypes.User(self.runt, udef['iden'])
+
+    async def _storPerms(self, perms):
+        s_stormtypes.confirm(('storm', 'lib', 'cortex', 'httpapi', 'set'))
+        perms = await s_stormtypes.toprim(perms)
+        pdefs = []
+        for pdef in perms:
+            if isinstance(pdef, str):
+                if pdef.startswith('!'):
+                    raise s_exc.BadArg(mesg=f'Permission assignment must not start with a !, got {pdef}')
+                parts = pdef.split('.')
+                pdef = {'perm': parts, 'default': False}
+
+            pdefs.append(pdef)
+        await self.runt.snap.core.modHttpExtApi(self.iden, 'perms', pdefs)
+        self.info['perms'] = pdefs
+
+    @s_stormtypes.stormfunc(readonly=True)
+    async def _gtorPerms(self):
+        return self.info.get('perms')
```

Comment:
Do we want a perms wrapper that would allow someone to do the following?

```
$api = $lib.cortex.httpapi.get($iden)
$api.perms.append(your.new.perm)
```

Reply:
If things like vars can be mutated, we should probably make this support that as well.